torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -721,20 +722,22 @@ std::vector<at::Tensor> FusionExecutor::allocOutputs(
   for (const auto out_i : c10::irange(kernel->outputs().size())) {
     // TODO: FIX this short-cut where we trivially forward inputs to outputs
     if (kernel->outputs()[out_i]->isFusionInput()) {
-      TORCH_INTERNAL_ASSERT(false, "trivial input forwarding NOT IMPLEMENTED");
-      // for (auto inp_i : c10::irange(kernel->inputs().size())) {
-      //   if (kernel->inputs()[inp_i] == kernel->outputs()[out_i]) {
-      //     TORCH_INTERNAL_ASSERT(
-      //         inp_i < inputs.size(),
-      //         "Issue with an input showing up as output, couldn't find
-      //         input.");
-      //     TORCH_INTERNAL_ASSERT(
-      //         inputs[inp_i].isTensor(),
-      //         "Cannot register a scalar as an output in a fusion.");
-      //     outputs.push_back(inputs[inp_i].toTensor());
-      //     break;
-      //   }
-      // }
+      for (auto inp_i : c10::irange(kernel->inputs().size())) {
+        if (kernel->inputs()[inp_i] == kernel->outputs()[out_i]) {
+          TORCH_INTERNAL_ASSERT(
+              inp_i < args.size(),
+              "Issue with an input showing up as output, couldn't find input.");
+          TORCH_INTERNAL_ASSERT(
+              args[inp_i]->isType(ArgType::Tensor),
+              "Cannot register a scalar as an output in a fusion.");
+          // pushing empty tensor for trivial forwarding. Since we handle this in integration
+          c10::Device device(c10::DeviceType::CUDA, args.getDeviceIndex());
+          const auto tensor_options =
+              at::TensorOptions().dtype(at::kFloat).device(device);
+	  outputs.emplace_back(at::empty({0}, tensor_options));
```

Comment:
We want a place holder so that we are not allocating anything for trivially forwarded inputs. Basically, codegen is ignoring all trivially forwarded output, because there's no producer of those and we are passing an empty tensor on IO just so we'll match kernel signature.

Later when we put together the output of the fusion, forwarded inputs are copied (only meta tensor) directly to the output.

There's some funny bits here between the interaction of `alias` vs `trivially-forwarded`. I don't think we have a fool-proof API for the two at the moment. We can clean up afterwards. I'm enabling the support here since Ivan needed it for his copy_to PR. https://github.com/pytorch/pytorch/pull/84545

Reply:
Thanks for the early review. I'll put some doc in code when I get it cleaned up.